pymongo/asynchronous/client_bulk.py

```diff
@@ -133,7 +134,9 @@ def add_insert(self, namespace: str, document: _DocumentOut) -> None:
         validate_is_document_type("document", document)
         # Generate ObjectId client side.
         if not (isinstance(document, RawBSONDocument) or "_id" in document):
-            document["_id"] = ObjectId()
+            document = ChainMap(document, {"_id": ObjectId()})
+        elif not isinstance(document, RawBSONDocument) and "_id" in document:
```

Comment:
Could you refactor these two if statements to avoid repeating the checks?
```python
if not isinstance(document, RawBSONDocument):
    if "_id" in document:
        ...
    else:
```

Reply:
It would also be worthwhile to add a comment explaining the ChainMap usage and why we have to use here but not in other insert code paths.